### PR TITLE
Feature/STACKn-197

### PR DESCRIPTION
**Pull request by adamhagevall**
## Status

- [ ] Ready
- [x] Draft
- [ ] Hold

## Description

Added a CLI command "stackn run" that enables users to run a complete ML pipeline as defined in a JSON object called pipeline.json in the initialised project structure. The command "stackn train" is now redundant; if the user only wants to run the training script, they can include this and nothing else in pipeline.json as explained above.  

Changes have been made where appropriate to suit the new command. This includes changes in the files stackn_cmd.py, StudioClient.py, TrackingClient.py, models_details.html, and models/views.py. 

When logging with the TrackingClient, users must now use the newly implemented class function define_operation() to state what pipeline operation (e.g. data preprocessing, training, validation, prediction) that the current logging operation refers to. For example, when logging data from the training script, the user must use something like TrackingClient().define_operation("Training") as this will be used to label the local file where this data will be stored. It is also important to label the logged data so that it's easy to identify its origin. For example, when logging model accuracy during training, the metric should be labelled "Training accuracy", while model accuracy from validation must be labelled something like "Validation accuracy". 

## Types of changes

What types of changes does your code introduce to STACKn?

- [ ] Hotfix (fixing a critical bug in production)
- [ ] Bugfix
- [x] New feature
- [ ] Documentation update

## Checklist

_If you're unsure about any of the items below, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [ ] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have included migration files (if there are changes to the model classes)
- [x] I have read the [CONTRIBUTING](https://github.com/scaleoutsystems/stackn/blob/master/CONTRIBUTING.md) doc
- [ ] I have included tests to complement my changes
- [ ] I have updated the related documentation (if necessary) 
- [ ] I have updated the release notes (docs/releasenotes.md)
- [ ] I have added a reviewer for this pull request
- [ ] I have added myself as an author for this pull request

## Further comments

Anything else you think we should know before merging your code!

### CODE REVIEW

**Review by morganekmefjord — CHANGES_REQUESTED** [review]
You can exclude migrations as they are relative to your baseline db. In other deployments it may differ depending on which version is currently running. Our startup does "make migrations" which will calculate and apply the necessary model operations.

**Review by morganekmefjord — APPROVED** [review]
Nice implementation 👏

Side note;
There where a few places where you optimistically index dicts where I would prefer to see exception handling encapsulating then.

**Comment by dstoyanova** [comment]
> You can exclude migrations as they are relative to your baseline db. In other deployments it may differ depending on which version is currently running. Our startup does "make migrations" which will calculate and apply the necessary model operations.

We have actually changed this and the migration files are needed. The setup with making and applying migrations didn't work before and we couldn't figure out why so now we just apply migrations (without make). If these changes are to be in develop branch, they need to be submitted so that the automatic setup works when deploying.
